src/Search/Microsoft.Azure.Search/Customizations/DataSources/Models/DataSource.Customization.cs

```diff
@@ -0,0 +1,275 @@
+        {
+            return AzureSql(name, sqlConnectionString, tableOrViewName, deletionDetectionPolicy, description);
+        }
+
+        /// <summary>
+        /// Creates a new DataSource to connect to a VM-hosted SQL Server database with change detection enabled.
+        /// </summary>
+        /// <param name="name">The name of the datasource.</param>
+        /// <param name="sqlConnectionString">The connection string for the SQL Server database.</param>
+        /// <param name="tableOrViewName">The name of the table or view from which to read rows.</param>
+        /// <param name="changeDetectionPolicy">The change detection policy for the datasource.</param>
+        /// <param name="description">Optional. Description of the datasource.</param>
+        /// <returns>A new DataSource instance.</returns>
+        public static DataSource SqlServerOnAzureVM(
+            string name,
+            string sqlConnectionString,
+            string tableOrViewName,
+            DataChangeDetectionPolicy changeDetectionPolicy,
+            string description = null)
+        {
+            return AzureSql(name, sqlConnectionString, tableOrViewName, changeDetectionPolicy, description);
+        }
+
+        /// <summary>
+        /// Creates a new DataSource to connect to a VM-hosted SQL Server database with change detection and deletion detection enabled.
+        /// </summary>
+        /// <param name="name">The name of the datasource.</param>
+        /// <param name="sqlConnectionString">The connection string for the SQL Server database.</param>
+        /// <param name="tableOrViewName">The name of the table or view from which to read rows.</param>
+        /// <param name="changeDetectionPolicy">The change detection policy for the datasource. Note that only high watermark change detection is
+        /// allowed for SQL Server when deletion detection is enabled.</param>
+        /// <param name="deletionDetectionPolicy">The data deletion detection policy for the datasource.</param>
+        /// <param name="description">Optional. Description of the datasource.</param>
+        /// <returns>A new DataSource instance.</returns>
+        public static DataSource SqlServerOnAzureVM(
+            string name,
+            string sqlConnectionString,
+            string tableOrViewName,
+            HighWaterMarkChangeDetectionPolicy changeDetectionPolicy,
+            DataDeletionDetectionPolicy deletionDetectionPolicy,
+            string description = null)
+        {
+            return AzureSql(name, sqlConnectionString, tableOrViewName, changeDetectionPolicy, deletionDetectionPolicy, description);
+        }
+
+        /// <summary>
+        /// Creates a new DataSource to connect to a DocumentDb database.
+        /// </summary>
+        /// <param name="name">The name of the datasource.</param>
+        /// <param name="documentDbConnectionString">The connection string for the DocumentDb database.</param>
```

Comment:
It's a flavor for DocDb connection string that includes Database value (which is optional for DocDb connection strings). It would be useful to include an example here.